content/en/security/application_security/enabling/compatibility/java.md

```diff
@@ -6,45 +6,139 @@ type: multi-code-lang
 code_lang_weight: 0
 ---
 
+## ASM capabilities
+
+The following ASM capabilities are supported in the Java library, for the specified tracer version:
+
+| ASM capability                   | Minimum Java tracer version |
+| -------------------------------- | ----------------------------|
+| Threat Detection | 1.8.0  |
+| Threat Protection| 1.9.0 |
+| Open Source Vulnerability detection | 1.1.4 |
+| Custom Code Vulnerability detection | 1.15.0|
+
+The minimum tracer version to get all supported ASM capabilities for Java is 1.15.0.
+
+**Note**: Threat Protection requires enabling [Remote Configuration][2], which is included in the listed minimum tracer version.      
+
+### Supported deployment types
+|Type           | Threats support   |  Vulnerabilities support  |
+| ---           |   ---             |           ----            |
+| Docker        | {{< X >}}         | {{< X >}}                 |
+| Kubernetes    | {{< X >}}         | {{< X >}}                 | 
+| AWS ECS       | {{< X >}}         | {{< X >}}                 |
+| AWS Fargate   | {{< X >}}         | {{< X >}}                 |
+| AWS Lambda    | {{< X >}}         |                           |   
+
+
 ## Language and framework compatibility
 
 ### Supported Java versions
+The Java Tracer supports automatic instrumentation for the following Oracle JDK and OpenJDK JVM runtimes.
 
-The Datadog library supports Java JRE 1.8 and higher of both Oracle JDK and OpenJDK, on the following architectures:
-- Linux (GNU) x86, x86-64
-- Alpine Linux (musl) x86, x86-64
-- macOS (Darwin) x86, x86-64
-- Windows (msvc) x86, x86-64
+| JVM versions | Operating Systems                                                               | Support level                       | Tracer version |
+| -------------| ------------------------------------------------------------------------------- | ----------------------------------- | -------------- |
+| 8 to 17      | Windows (x86, x86-64)<br>Linux (x86, x86-64)<br>Mac (x86, x86-64)               | Supported                | Latest         |
```

Comment:
* It should probably be `macOS` instead of `Mac`.
* I would keep the explicit mention to glibc and musl in Linux.
* For Threats: we do not support x86. Supported architecture in Windows is only x86-64. Linux and macOS are supported on x86-64 and arm64.
* For Vulnerabilities: support is architecture-independent and should work anywhere where the JDK also works.

Reply:
Updated the java page with these suggestions, let me know if it looks better or if any other changes are needed